polyfill/index.d.ts

```diff
@@ -0,0 +1,385 @@
+    constructor(month: number, day: number);
+    readonly month: number;
+    readonly day: number;
+    with(temporalMonthDayLike: MonthDayLike, options?: AssignmentOptions): Temporal.MonthDay;
+    withYear(year: number | { year: number }): Temporal.Date;
+    getFields(): Required<MonthDayLike>;
+    toLocaleString(locales?: string | string[], options?: Intl.DateTimeFormatOptions): string;
+    toJSON(): string;
+    toString(): string;
+  }
+
+  export type TimeLike = {
+    hour?: number;
+    minute?: number;
+    second?: number;
+    millisecond?: number;
+    microsecond?: number;
+    nanosecond?: number;
+  };
+
+  export class Time extends Required<TimeLike> {
+    static from(item: Temporal.Time | string | object, options?: AssignmentOptions): Temporal.Time;
+    static compare(one: Temporal.Time, two: Temporal.Time): ComparisonResult;
+    constructor(
+      hour?: number,
+      minute?: number,
+      second?: number,
+      millisecond?: number,
+      microsecond?: number,
+      nanosecond?: number
+    );
+    readonly hour: number;
+    readonly minute: number;
+    readonly second: number;
+    readonly millisecond: number;
+    readonly microsecond: number;
+    readonly nanosecond: number;
+    with(temporalTimeLike: TimeLike, options?: AssignmentOptions): Temporal.Time;
+    plus(temporalDurationLike: DurationLike, options?: ArithmeticOptions): Temporal.Time;
+    minus(temporalDurationLike: DurationLike, options?: ArithmeticOptions): Temporal.Time;
+    // TODO: should 'days' be included?  Or would that be false advertising? See #580.
+    difference(other: Temporal.Time, options?: DifferenceOptions<'hours' | 'minutes' | 'seconds'>): Temporal.Duration;
+    withDate(temporalDate: Temporal.Date): Temporal.DateTime;
+    getFields(): Required<TimeLike>;
+    toLocaleString(locales?: string | string[], options?: Intl.DateTimeFormatOptions): string;
+    toJSON(): string;
+    toString(): string;
+  }
+
+  export type TimeZoneLike = Temporal.TimeZone | string;
+
+  export class TimeZone {
+    static from(timeZone: TimeZoneLike): Temporal.TimeZone;
+    constructor(timeZoneIdentifier: string);
+    readonly name: string;
+    getOffsetNanosecondsFor(absolute: Temporal.Absolute): number;
+    getOffsetStringFor(absolute: Temporal.Absolute): string;
+    getDateTimeFor(absolute: Temporal.Absolute): Temporal.DateTime;
+    getAbsoluteFor(dateTime: Temporal.DateTime, options?: ToAbsoluteOptions): Temporal.Absolute;
+    // TODO: should this be changed to getNextTransition/getPreviousTransition? See #613.
+    getTransitions(startingPoint: Temporal.Absolute): IteratorResult<Temporal.Absolute>;
+    toString(): string;
+    toJSON(): string;
+  }
+
+  export type YearMonthLike = {
+    year?: number;
+    month?: number;
+  };
+
+  export class YearMonth extends Required<YearMonthLike> {
+    static from(item: string | object, options?: AssignmentOptions): Temporal.YearMonth;
```

Comment:
`YearMonthLike` too?

Reply:
Yep, will fix... and will think twice in the future about using search-n-replace for anything other than renaming!